### DISCUSSION TITLE
Approach for simplifying Netbox UI/UX for certain group of users

### DISCUSSION THREAD

**Discussion by dejantep** [netbox-community/netbox#15406] [General]
Let's say i would like users to be able to create/assign ip-addresses or add virtual interfaces to devices and so on BUT i would like them to use a much simpler UI. Netbox can be overwhelming at times.
For instance you want to assign an ip-address... pick from list of prefixes and assign/create ip. only couple of fields, straightforward
Behind that "form" everything would behave as normal just fewer fields, bare minimum.
Think of onboarding plugin.

Would this be better off with a plugin or some kind of external web application that talks Netbox API?

I'm thinking..plugin would be nice if you want users working within Netbox but it seems more complicated as compared to a web app

**Comment by candlerb** [comment]
Those are reasonable approaches. Here are a couple of other ideas:
1. If you create a user with restricted permissions, that will simplify the Netbox UI to a degree (any objects they don't have read access to, won't appear in the menu at all)
2. You can create Custom Scripts to do specific tasks, driven by a form input - you could consider them a sort of single page "wizard". These will integrate directly with the Django ORM, so will be a bit like writing a plugin, but more restricted (and hence simpler).

But otherwise you may be better off with an external application that you control the look and feel of. Netbox does not provide a complete UI/UX customization environment in the way that Salesforce does, for example.

**Reply by DanSheps** [reply]
100% agree with setting up scripts to do the tasks.

I do this for our facilities and a few other teams.
